### PR TITLE
fix: formatting issues in order invoice

### LINKED ISSUE #8160: Ticket Invoice: Do not show "USD" for free tickets and fix currency number formatting to be equal everywhere on the invoice
For free tickets invoices show USD. 

1. Instead they should not show any currency for of the amount is 0.
2. Tickets no matter what the rate is should show two numbers after the dot, e.g. USD 12.50 instead of 12.5.
3. Add a space between USD [space] number. Make it the same with all numbers.

Screenshot from 2021-08-09 14-11-44

Screenshot from 2021-08-09 14-11-23

### PR DESCRIPTION
**Pull request by pc-beast**
Fixes #8160

#### Screeshot:
image

#### Short description of what this resolves:
There were some formatting issues in the order invoice as described in the issue. 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [x] The unit tests pass locally with my changes 
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

- [ ] All the functions created/modified in this PR contain relevant docstrings.

### CODE REVIEW

File: app/templates/pdf/order_invoice.html [hunk 1]
```diff
@@ -218,19 +223,37 @@ <h1>{{ _('INVOICE') }}</h1>
...
+                    {{ event.payment_currency }}&nbsp;
+                  {% endif %}
+                  {{ tax.rate*order_ticket.ticket.price/100 }}
+                </td>
             {% else %}
-                <td class="total">{{ event.payment_currency }}0</td>
+                <td class="total">
+                  {% if order_ticket.ticket.price %}
+                    {{ event.payment_currency }}&nbsp;
+                  {% endif %}
```
Comment (maze-runnar) [inline app/templates/pdf/order_invoice.html:254]: why add this? if below-mentioned amount is already 0.00

Comment (pc-beast) [reply app/templates/pdf/order_invoice.html:254]: right

File: app/templates/pdf/order_invoice.html [hunk 2]
```diff
@@ -240,7 +260,12 @@ <h1>{{ _('INVOICE') }}</h1>
             <td></td>
             <td></td>
             <td class="grand total"><b>{{ _('Grand Total') }}</b></td>
-            <td class="grand total"> {{ event.payment_currency }}{{ order.amount | money }}</td>
+            <td class="grand total"> 
+              {% if order.amount %}
```
Comment (maze-runnar) [inline app/templates/pdf/order_invoice.html:264]: Above you checked order.amount !=0 and here only order.amount, any specific reason?

**Review by maze-runnar — APPROVED** [review]

File: app/templates/pdf/order_invoice.html (hunk 2 above)
Comment (pc-beast) [reply app/templates/pdf/order_invoice.html]: Thank you for editing it. I was not available today but I got the point. It should have been consistent.

**Pull request merged**